### PR TITLE
private cells + Semantic Versioning

### PR DESCRIPTION
**Pull request by stoffern**
this closes https://github.com/stoffern/pokemongo-api/issues/52
@jimmywarting will this cover it?

### CODE REVIEW

**Comment by gallexme** [comment]
@stoffern  almost 
@jimmywarting   wants the forts/gyms be sorted by distance aswell

**Comment by stoffern** [comment]
added sort by distance

**Comment by stoffern** [comment]
also adding Travis IO so we can have better versioning.
solvs https://github.com/stoffern/pokemongo-api/issues/49

**Comment by jimmywarting** [comment]
hmm, why are we adding lastDistance? isn't just distance enough?

File: src/index.js [hunk 1]
```diff
@@ -95,24 +114,55 @@ class PokemonGOAPI {
...
+          objects.forts.checkpoints.push(fort)
+        else
+          objects.forts.gyms.push(fort)
+        }
+      )
+      //sort checkpoints
+      objects.forts.checkpoints.sort(dynamicSort("lastDistance"));
+      objects.forts.gyms.sort(dynamicSort("lastDistance"));
     }
     this.player.surroundings = cells
```
Comment (jimmywarting) [inline src/index.js]: surroundings should be equal to objects

**Comment by stoffern** [comment]
@dougiefresh49 Travis is failing on build `s2geometry-node` is it posslible to ignore packages?

**Comment by stoffern** [comment]
@jimmywarting that should do it.. we just need Travis to work..

**Comment by gallexme** [comment]
@stoffern 
@jimmywarting 
also experiencing 100% cpu usage caused by the get_distance from forts?

**Comment by gallexme** [comment]
http://puu.sh/qhGmi/7435a779b6.png

**Comment by jimmywarting** [comment]
hmm

**Comment by jimmywarting** [comment]
Found out why, and it's fixed now.
@stoffern sorted the array in every cell loop... only need to do it once

**Comment by gallexme** [comment]
im still experiencing 100%cpu usage 

i think that geolib.getDistance is to accurate for our needs
as we dont really have to care about earth curvature we may should just use a faster algorithm with less precision?

**Comment by jimmywarting** [comment]
My Money is on some faster algorithm. Belkin dose a lot of stuff

Maybe the second answer here: http://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude

**Comment by jimmywarting** [comment]
The distance getter fn is not called so often that it's not worth setting the distance once because it's only called once?

Beside I would like to get a updated distance if my location change

**Comment by gallexme** [comment]
@jimmywarting 
https://github.com/stoffern/pokemongo-api/blob/31e19a66824d6d0d83a24e03a832bee58b5d6c11/src/index.js#L247

if we have like 50 pokestops u calling it over 100 times

**Comment by gallexme** [comment]
@jimmywarting yeah would be great if you would try that algorithm its way way way less math then in geolib

**Comment by jimmywarting** [comment]
Why don't you give it a shot @gallexme?
try both method (the other algorithm and also setting the distance once)

**Comment by gallexme** [comment]
sure as soon as im back form work :)

**Pull request merged**